### PR TITLE
book: update concepts pages

### PR DESCRIPTION
**Pull request by ahayzen-kdab**

### CODE REVIEW

File: book/src/concepts/generated_qobject.md [hunk 1]
```diff
@@ -7,100 +7,39 @@ SPDX-License-Identifier: MIT OR Apache-2.0
...
 Whilst this QObject is a C++ type, CXX-Qt will automatically wrap it as a [CXX Opaque Type](https://cxx.rs/extern-c++.html#opaque-c-types).
-These generated QObjects are accessible to Rust in a generated module with the name `qobject`. Each struct `T`'s generated QObject is accessible as `qobject::T`.
+
+> If the bridge module is named `qobject`, then the C++ type can be reached via `qobject::T`
 
 ## Anatomy
 
 Any QObject generated by CXX-Qt is just a C++ QObject subclass that owns an instance of the Rust struct.
 The instance of the Rust struct is constructed using its [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) implementation in the C++ constructor.
 Therefore implementing `Default` on the Rust struct is currently mandatory.
```
Comment (LeonMatthesKDAB) [inline book/src/concepts/generated_qobject.md:23]: This is no longer true. You can use `Constructor` instead.

Comment (LeonMatthesKDAB) [inline book/src/concepts/generated_qobject.md:23]:
```suggestion
By default the instance of the Rust struct is constructed using the [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) trait.
If the Rust struct cannot implement `Default`, providing a custom constructor with the [`Constructor` trait](https://docs.rs/cxx-qt/latest/cxx_qt/trait.Constructor.html) is required.
```

File: book/src/concepts/generated_qobject.md [hunk 2]
```diff
@@ -7,100 +7,39 @@ SPDX-License-Identifier: MIT OR Apache-2.0
...
+
+For example if the bridge module was called `qobject` and the type was called `T` an impl block would be written as `impl qobject::T { ... }`.
+
+> Methods from traits, such as [Threading](https://docs.rs/cxx-qt/latest/cxx_qt/trait.Threading.html), are available in the C++ context
+
+> From a C++ context the Rust context can be reach by using methods on the [CxxQtType](https://docs.rs/cxx-qt/latest/cxx_qt/trait.CxxQtType.html) trait
+
+## Rust context
+
+The only requirement for the Rust struct is that it has a `Default` or a [cxx_qt::Constructor](https://docs.rs/cxx-qt/latest/cxx_qt/trait.Constructor.html) implementation.
```
Comment (LeonMatthesKDAB) [inline book/src/concepts/generated_qobject.md:43]:
```suggestion
The only requirement for the Rust struct is that it has a `Default` or that the QObject implements [cxx_qt::Constructor](https://docs.rs/cxx-qt/latest/cxx_qt/trait.Constructor.html).
```

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
Some small errors regarding `cxx_qt::Constructor`.
Otherwise looks good.

File: book/src/concepts/generated_qobject.md (hunk 1 above)
Comment (ahayzen-kdab) [reply book/src/concepts/generated_qobject.md:23]: done

File: book/src/concepts/generated_qobject.md (hunk 2 above)
Comment (ahayzen-kdab) [reply book/src/concepts/generated_qobject.md:43]: done

File: book/src/concepts/generated_qobject.md (hunk 1 above)
Comment (ahayzen-kdab) [reply book/src/concepts/generated_qobject.md:23]: the next comment was the change to fix this I assume ?

Comment (LeonMatthesKDAB) [reply book/src/concepts/generated_qobject.md]: Yup, sorry for the duplicate.

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**